### PR TITLE
Add zapzap application

### PR DESCRIPTION
**Pull request by hatch01**
## Description of changes

Just add the app zapzap to nixpkgs

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ x ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/zapzap/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+, buildPythonApplication
+, setuptools
+, pyqt6
+, pyqt6-webengine
+, dbus-python
+}:
+
+buildPythonApplication rec {
+  pname = "zapzap";
+  version = "4.5.5.2";
```
Comment (WolfangAukang) [inline pkgs/applications/networking/instant-messengers/zapzap/default.nix:12]:
```suggestion
  version = "4.5.5.2";
  format = "setuptools";
```

File: pkgs/applications/networking/instant-messengers/zapzap/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  propagatedBuildInputs = [
+    setuptools
+    dbus-python
+    pyqt6
+    pyqt6-webengine
+  ];
+
+  prePatch = "export HOME=$NIX_BUILD_TOP";
```
Comment (WolfangAukang) [inline pkgs/applications/networking/instant-messengers/zapzap/default.nix:28]: Why is this being set? Please specify on a comment above the line.

File: pkgs/applications/networking/instant-messengers/zapzap/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  installPhase = ''
+    python setup.py install --prefix=$out
+    mkdir -p $out/usr/share/applications
+    cp -R share/applications/com.rtosta.zapzap.desktop $out/usr/share/applications/com.rtosta.zapzap.desktop
+    mkdir -p $out/usr/share/icons/hicolor/scalable/apps/
+    cp -R share/icons/com.rtosta.zapzap.svg $out/usr/share/icons/hicolor/scalable/apps/com.rtosta.zapzap.svg
+  '';
+
+  meta = with lib; {
```
Comment (WolfangAukang) [inline pkgs/applications/networking/instant-messengers/zapzap/default.nix:38]: Meta can have a `changelog` (based on the GitHub releases) and `mainProgram` sections

**Review by WolfangAukang — CHANGES_REQUESTED** [review]
Thank you for the contribution, but you definitely didn't follow the [contributing guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) at all. Besides the specified changes, consider the following:
- Commits should follow the naming convention specified on the contributing guidelines
- Adding a new maintainer should be in a single commit
- The package addition should be in a single commit
- Please consider adding the package to `pkgs/by-name` instead of the current directory - see [info](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md)

**Comment by drupol** [comment]
I'm locking and closing this pull request, because through the rebase you inadvertently requested many maintainers for review, which subscribed them to update notifications, resulting in unnecessary spam inside everyone's inbox. While we can remove their review requests, we sadly cannot unsubscribe anyone.

Please create a new pull request and for the next time remember to set your PR to draft status before rebasing. In draft status, you can preview the list of maintainers that are about to be requested for review, which allows you to sidestep this issue.

**Pull request closed without merging**